Microsoft defender for cloud collector (#23631)
demisto/content: Packs/AzureSecurityCenter/Integrations/MicrosoftDefenderForCloudEventCollector/README.md

XSIAM collector for Microsoft Defender for Cloud alerts.

> This integration was integrated and tested with version xx of Microsoft Defender for Cloud Event Collector

Reviewer: Replace the xx with a real version

Author: deleted this line.

## Configure Microsoft Defender for Cloud Event Collector on Cortex XSIAM

1. Navigate to **Settings** > **Configurations** > **Data Collection** > **Automations & Feed Integrations**.
2. Search for Microsoft Defender for Cloud Event Collector.
3. Click **Add instance** to create and configure a new integration instance.

    | **Parameter** | **Description** | **Required** |
    | --- | --- | --- |
    | Microsoft Azure Management URL |  | False |
    | ID |  | True |
    | Token |  | True |
    | Key |  | True |
    | Certificate Thumbprint | Used for certificate authentication. As appears in the "Certificates & secrets" page of the app. | False |
    | Private Key | Used for certificate authentication. The private key of the registered certificate. | False |
    | Subscription ID to use |  | True |
    | Trust any certificate (not secure) |  | False |

>     | Use system proxy settings |  | False |

Reviewer: Add missing description

Author: This is auto-generated from the yml added description where it is not self-explanatory.

XSIAM collector for Microsoft Defender for Cloud alerts.
This integration was integrated and tested with version xx of Microsoft Defender for Cloud Event Collector

> ## Configure Microsoft Defender for Cloud Event Collector on Cortex XSOAR

Reviewer 2: ```suggestion
## Configure Microsoft Defender for Cloud Event Collector on Cortex XSIAM
```

> 1. Navigate to **Settings** > **Integrations** > **Servers & Services**.

Reviewer 2: ```suggestion
1. Navigate to **Settings** > **Configurations** > **Data Collection** > **Automations & Feed Integrations**.
```

2. Search for Microsoft Defender for Cloud Event Collector.
3. Click **Add instance** to create and configure a new integration instance.

    | **Parameter** | **Description** | **Required** |
    | --- | --- | --- |
    | Microsoft Azure Management URL |  | False |
    | ID |  | True |
    | Token |  | True |
    | Key |  | True |

>     | Certificate Thumbprint | Used for certificate authentication. As appears in the "Certificates &amp;amp; secrets" page of the app. | False |

Reviewer 2: Need to fix &amp;amp;
What is this supposed to be?

Author: fixed Thanks.

    | Private Key | Used for certificate authentication. The private key of the registered certificate. | False |
    | Subscription ID to use |  | True |
    | Trust any certificate (not secure) |  | False |
    | Use system proxy settings |  | False |

4. Click **Test** to validate the URLs, token, and connection.

## Commands

> You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.

Reviewer 2: Should this be the Cortex XSIAM CLI?

Author: good catch changed it thanks

1. Navigate to **Settings** > **Configurations** > **Data Collection** > **Automations & Feed Integrations**.
2. Search for Microsoft Defender for Cloud Event Collector.
3. Click **Add instance** to create and configure a new integration instance.

    | **Parameter** | **Description** | **Required** |
    | --- | --- | --- |
    | Microsoft Azure Management URL |  | False |
    | Client ID | Microsoft Defender for Cloud client ID | True |
    | Tenant ID | Microsoft Defender for Cloud Tenant ID | True |
    | Client Secret | Microsoft Defender for Cloud Client Secret | True |
    | Certificate Thumbprint | Used for certificate authentication. As appears in the "Certificates &amp;amp; secrets" page of the app. | False |
    | Private Key | Used for certificate authentication. The private key of the registered certificate. | False |
    | Subscription ID to use |  | True |
    | First fetch time interval | First time to start fetching alerts from. | False |
    | Trust any certificate (not secure) |  | False |
    | Use system proxy settings |  | False |

4. Click **Test** to validate the URLs, token, and connection.

## Commands

> You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.

Reviewer 2: ```suggestion
You can execute these commands from the Cortex XSIAM CLI, as part of an automation, or in a playbook.
```

After you successfully execute a command, a DBot message appears in the War Room with the command details.

## Known limitations
This integration does not have a fetch limit parameter due to the limitations of the API functionality.
1. The collector fetches all events between the current time and the last time it was fetched during every fetch operation.
2. If the command is run for the first time, all events from ***first_fetch*** to now will be fetched in one execution.

> It is possible that the cause of 1 and 2 may cause the fetch to take some time. You may need to increase the collector time out in the server configuration if the collector fetch times out.

Reviewer 2: ```suggestion
It is possible that the above limitations may cause the fetch to take some time. You may need to increase the collector time out value in the server configuration if the collector fetch times out.
```
